cgi-bin/DW/Controller/Entry.pm

```diff
@@ -740,12 +740,13 @@ sub _form_to_backend {
     $props->{taglist} = "" unless $props->{taglist} && $props->{taglist} =~ /\S/;
 
     if ( LJ::is_enabled( 'adult_content' ) ) {
+        my $restriction_key = $post->{age_restriction} || '';
```

Comment:
For this kind of usage, where you've got `undef` on the left side, you need to use `//` instead of `||`. The former operator checks for defined-ness, which means you don't accidentally squash false values.